source/common/http/conn_manager_impl.h

```diff
@@ -290,14 +290,16 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
   void onEvent(Network::ConnectionEvent event) override;
   // Pass connection watermark events on to all the streams associated with that connection.
   void onAboveWriteBufferHighWatermark() override {
-    for (ActiveStreamPtr& stream : streams_) {
-      stream->callHighWatermarkCallbacks();
-    }
+    ASSERT(codec_);
```

Comment:
nit: this will crash in an obvious way a few lines later so I would kill this assert. Same below.